plugins/rbac/src/hooks/useRole.ts

```diff
@@ -0,0 +1,31 @@
+import { useAsync } from 'react-use';
+
+import { useApi } from '@backstage/core-plugin-api';
+
+import { Role } from '@janus-idp/backstage-plugin-rbac-common';
+
+import { rbacApiRef } from '../api/RBACBackendClient';
+
+export const useRole = (
+  roleEntityRef: string,
+): {
+  loading: boolean;
+  role: Role | undefined;
+  roleError: Error;
+} => {
+  const rbacApi = useApi(rbacApiRef);
+  const {
+    value: roles,
+    loading,
+    error: roleError,
+  } = useAsync(async () => await rbacApi.getRole(roleEntityRef));
+
+  return {
+    loading,
+    role: Array.isArray(roles) ? roles[0] : undefined,
+    roleError: (roleError as Error) || {
+      name: (roles as Response)?.status,
+      message: `Error fetching the role. ${(roles as Response)?.statusText}`,
+    },
```

Comment:
with this data and roleError will always be there i.e as below, ideally it should be there only if there is error.

when data is there, roleError looks like below

```
message: "Error fetching the role. undefined"
name:  undefined
```

Above change is not related to this PR so we can take this as an issue and get this improved

cc @divyanshiGupta